lib/mountain_view/presenter.rb

```diff
@@ -0,0 +1,70 @@
+module MountainView
+  class Presenter
+    class_attribute :_attributes, instance_accessor: false
+    self._attributes = {}
+
+    attr_reader :slug, :properties
+    alias_method :props, :properties
+
+    def initialize(slug, properties={})
+      @slug = slug
+      @properties = attribute_defaults.deep_merge(properties)
+    end
+
+    def partial
+      "#{slug}/#{slug}"
+    end
+
+    def locals
+      locals = build_hash
+      locals.merge(properties: locals)
+    end
+
+    private
+
+    def attributes
+      @attributes ||= properties.keys.inject({}) do |sum, name|
+        sum[name] = {}
```

Comment:
Why is the default a hash here?
Does it break stuff if we just don't provide a value or make it `nil`?